### PR TITLE
Replace filterable with translatable, and add supoort for date …

### PR DESCRIPTION
**Pull request by boyjoy1127**
Ref #19937.

Changes proposed in this pull request:
  - replace filterable with translatable.
  - add support for date type column.

---

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [X] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [X] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Hi @boyjoy1127, can you fix ci error?

File: kernel/sql-federation/optimizer/src/main/antlr4/org/apache/shardingsphere/sqlfederation/optimizer/parser/rexnode/Literals.g4 [hunk 1]
```diff
@@ -43,3 +47,14 @@ INT_
...
+YEAR_
+    : [0-9]+
+    ;
+
+MONTH_
+    : [0-9]+
+    ;
+DAY_
+    : [0-9]+
+    ;
```
Comment (strongduanmu) [inline kernel/sql-federation/optimizer/src/main/antlr4/org/apache/shardingsphere/sqlfederation/optimizer/parser/rexnode/Literals.g4:60]: Please add a new line at the end of file.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by wgy8283335** [comment]
> Hi @boyjoy1127, can you fix ci error?

OK

File: infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/decider/engine/SQLFederationDeciderEngine.java [hunk 2]
```diff
@@ -62,6 +62,9 @@ public SQLFederationDeciderContext decide(final QueryContext queryContext, final
         SQLFederationDeciderContext result = new SQLFederationDeciderContext();
         SQLStatementContext<?> sqlStatementContext = queryContext.getSqlStatementContext();
         // TODO move this logic to SQLFederationDecider implement class when we remove sqlFederationEnabled
+        if (queryContext.getSql().contains("WHEN")) {
```
Comment (strongduanmu) [inline infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/decider/engine/SQLFederationDeciderEngine.java:65]: Why add this judgement?

File: test/e2e/suite/src/test/resources/cases/dql/dataset/db/select_sys_data_for_pg.xml [hunk 3]
```diff
@@ -22,22 +22,19 @@
...
-    <row values="public | t_order_item_federate_sharding | table | mock user" />
-    <row values="public | t_order_item_join_view | view | mock user" mayNotExist="true" />
-    <row values="public | t_order_subquery_view | view | mock user" mayNotExist="true" />
-    <row values="public | t_order_union_view | view | mock user" mayNotExist="true" />
-    <row values="public | t_product | table | mock user" />
-    <row values="public | t_product_category | table | mock user" />
-    <row values="public | t_product_detail | table | mock user" />
-    <row values="public | t_single_table | table | mock user" />
-    <row values="public | t_user | table | mock user" />
+    <row values="public | t_broadcast_table | table | postgres" />
```
Comment (strongduanmu) [inline test/e2e/suite/src/test/resources/cases/dql/dataset/db/select_sys_data_for_pg.xml:25]: Please support PostgreSQL \d statements with sql federation. If the current scenario is table sharding, the table name returned is incorrect.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/decider/engine/SQLFederationDeciderEngine.java (hunk 2 above)
Comment (wgy8283335) [reply infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/decider/engine/SQLFederationDeciderEngine.java:65]: The "WHEN" is not supported by the federation query.

File: test/e2e/suite/src/test/resources/cases/dql/dataset/db/select_sys_data_for_pg.xml (hunk 3 above)
Comment (wgy8283335) [reply test/e2e/suite/src/test/resources/cases/dql/dataset/db/select_sys_data_for_pg.xml:25]: The content of it-env.properties is below:
"it.scenarios=db"

How can I figure out the correct table name?

Comment (wgy8283335) [reply test/e2e/suite/src/test/resources/cases/dql/dataset/db/select_sys_data_for_pg.xml:25]: The content of it-env.properties is below:
"it.scenarios=db"

How can I figure out the correct table name?

**Comment by TeslaCN** [comment]
I'm moving this to 5.3.3.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**